dissect/target/plugins/os/unix/linux/_os.py

```diff
@@ -23,9 +24,22 @@ def detect(cls, target: Target) -> Optional[Filesystem]:
                 return fs
         return None
 
-    @export(property=True)
-    def ips(self) -> list[str]:
-        return self.network_manager.get_config_value("ips")
+    @export(property=False)
+    @arg("--dhcp", action="store_true", help="Parse syslogs for recent DHCP leases")
+    def ips(self, dhcp: bool = False) -> list[str]:
+        """Returns a list of IP addresses found on the host system."""
```

Comment:
We added the `--dhcp` option because of a previous comment from @Schamper in https://github.com/fox-it/dissect.target/pull/126#discussion_r1069117949.

Reply:
Yeah and I think that argument is still valid here, however here we have another issue which is that `ips` is a `property`. So now there's an API conflict where for some targets (Windows) you must do `t.ips` and for some (Unix) `t.ips()`, because only functions can take arguments.

In the future I'd like to completely restructure all network related stuff (cross-OS) to be able to more easily differentiate between network interfaces, static and dynamic IPs, routes, DNS, etc. However we're obviously not there yet.

I'm leaning towards moving this behaviour into a separate function for the time being (`ips_dhcp` perhaps?) with a note to fix this properly at a later stage when the previously mentioned idea comes to fruition.

Reply:
Added separate `ips_dhcp` property in [bcabac5](https://github.com/fox-it/dissect.target/pull/133/commits/bcabac56635db96dcd17408f6eb1dc9d4cdaa93d).